---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```


# GencoDymo2 <img src="https://img.shields.io/badge/version-1.0.1-blue.svg" align="right" />

[![Project Status: Active â€“ The project has reached a stable, usable state and is being actively developed.](https://www.repostatus.org/badges/latest/active.svg)](https://www.repostatus.org/#active)
[![Licence: GPL v3](https://img.shields.io/badge/license-GPL--3-blue.svg)](https://www.gnu.org/licenses/gpl-3.0)
[![Maintenance](https://img.shields.io/badge/Maintained%3F-yes-green.svg)](https://github.com/monahton)
[![GitHub issues](https://img.shields.io/github/issues/monahton/GencoDymo2)](https://github.com/monahton/GencoDymo2/issues)
[![Last commit](https://img.shields.io/github/last-commit/monahton/GencoDymo2)](https://github.com/monahton/GencoDymo2/commits/main)
[![Platform](https://img.shields.io/badge/platform-all-green)](https://cran.r-project.org/)
[![Website](https://img.shields.io/badge/docs-website-blue)](https://monahton.github.io/GencoDymo2/)

---

## ğŸ“¦ Overview

**GencoDymo2** is an R package tailored for dynamic extraction, exploration, and comparison of gene annotations from the [GENCODE](https://www.gencodegenes.org) database for human and mouse genomes. This tool facilitates:

- Automated retrieval of the latest or specific GENCODE releases
- Visualization and quantification of annotation changes across versions
- Extraction of introns, exons, splice motifs
- Generation of splice site FASTA files for tools like MaxEntScan

It bridges bioinformatics workflows and annotation dynamics, enhancing reproducibility and comparative studies in transcriptome and splicing research.

---

## ğŸ’» Installation

You can install the **development version** of `GencoDymo2` directly from GitHub:

```r
# install.packages("devtools")
devtools::install_github("monahton/GencoDymo2")
```

Then load it into your R session:

```r
library(GencoDymo2)
```

---

## ğŸš€ Getting Started

To get started, view the vignette:

```r
vignette("GencoDymo2")
```

Or visit the documentation website:  
ğŸ‘‰ https://monahton.github.io/GencoDymo2/

---

## ğŸ“ Functions Highlights

| Function              | Description                                                  |
|-----------------------|--------------------------------------------------------------|
| `get_latest_release()` | Retrieves the latest available GENCODE release per species |
| `compare_release()`   | compare annotation statistics between releases                              |
| `extract_introns()`    | Extracts and processes introns from annotation              |
| `assign_splice_sites()`     | Assign the donor and acceptor splice sites                         |
| `extract_ss_motif()` | Extract splicing motifs for MaxEntScan tool            |

---

## ğŸ› ï¸ Development & Contributing

**GencoDymo2** is actively developed. Contributions and suggestions are welcome!

- ğŸ”§ Open issues: https://github.com/monahton/GencoDymo2/issues
- ğŸ“¬ Email: <aboualezz.monah@hsr.it>
- ğŸ¤ Pull requests encouraged!

---

## :writing_hand: Author

**Monah Abou Alezz, PhD** -- <aboualezz.monah@hsr.it>.  

San Raffaele Telethon Institute for Gene Therapy (SR-TIGET)  
IRCCS San Raffaele Scientific Institute, Milan, Italy  

ğŸŒ [Personal website](https://monahton.github.io)

[![saythanks](https://img.shields.io/badge/say-thanks-ff69b4.svg)](https://saythanks.io/to/monahton)
`r badger::badge_custom("follow me on", "LinkedIn", "blue", "https://linkedin.com/in/monah-abou-alezz-phd-06a948ba")`

---

## ğŸ§ª Related Projects

- [GENCODE](https://www.gencodegenes.org)
- [MaxEntScan](http://hollywood.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html)

---

## ğŸ“£ Acknowledgments

Developed as part of ongoing research on lncRNA splicing and gene annotation evolution.  
Special thanks to colleagues at IGM-CNR and collaborators across splicing research projects.
