<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Assign intron donor and acceptor splice sites consensus — assign_splice_sites • GencoDymo2</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Assign intron donor and acceptor splice sites consensus — assign_splice_sites"><meta name="description" content="This function takes a data frame of intron coordinates and a genome sequence (ideally human or mouse) and returns a data frame with two additional columns for the donor and acceptor splice site consensus sequences.
It prepares the donor and acceptor sequences based on the provided intron coordinates and the specified genome (e.g., human hg38), making it useful for downstream analysis of splicing events."><meta property="og:description" content="This function takes a data frame of intron coordinates and a genome sequence (ideally human or mouse) and returns a data frame with two additional columns for the donor and acceptor splice site consensus sequences.
It prepares the donor and acceptor sequences based on the provided intron coordinates and the specified genome (e.g., human hg38), making it useful for downstream analysis of splicing events."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GencoDymo2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GencoDymo2_vignette.html">GencoDymo2</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/monahton/GencoDymo2/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Assign intron donor and acceptor splice sites consensus</h1>
      <small class="dont-index">Source: <a href="https://github.com/monahton/GencoDymo2/blob/main/R/introns.R" class="external-link"><code>R/introns.R</code></a></small>
      <div class="d-none name"><code>assign_splice_sites.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function takes a data frame of intron coordinates and a genome sequence (ideally human or mouse) and returns a data frame with two additional columns for the donor and acceptor splice site consensus sequences.
It prepares the donor and acceptor sequences based on the provided intron coordinates and the specified genome (e.g., human hg38), making it useful for downstream analysis of splicing events.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">assign_splice_sites</span><span class="op">(</span><span class="va">input</span>, genome <span class="op">=</span> <span class="va">BSgenome.Hsapiens.UCSC.hg38</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-input">input<a class="anchor" aria-label="anchor" href="#arg-input"></a></dt>
<dd><p>A data frame containing intron coordinates with the following columns:</p><ul><li><p><code>seqnames</code>: The chromosome name.</p></li>
<li><p><code>intron_start</code>: The start position of the intron.</p></li>
<li><p><code>intron_end</code>: The end position of the intron.</p></li>
<li><p><code>strand</code>: The strand on which the intron is located (<code>+</code> or <code>-</code>).</p></li>
<li><p><code>transcript_id</code>: The ID of the transcript to which the intron belongs.</p></li>
<li><p><code>intron_number</code>: The number of the intron within the transcript.</p></li>
<li><p><code>gene_name</code>: The name of the gene.</p></li>
<li><p><code>gene_id</code>: The gene ID.</p></li>
</ul></dd>


<dt id="arg-genome">genome<a class="anchor" aria-label="anchor" href="#arg-genome"></a></dt>
<dd><p>The genome sequence (BSgenome object) for the species. Default is the human genome (hg38).
This object is required for extracting the consensus sequences from the genome at the specified intron positions.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Logical. If TRUE, the function prints progress messages while preparing the splice site data. Default is TRUE.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A data frame containing the original intron data, with two additional columns:</p><ul><li><p><code>donor_ss</code>: The donor splice site consensus sequence for each intron.</p></li>
<li><p><code>acceptor_ss</code>: The acceptor splice site consensus sequence for each intron.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function performs the following steps:</p><ul><li><p>First, it prepares the splice site sequences for both donor and acceptor sites by calculating their positions based on the strand orientation and intron coordinates. The donor splice site is typically located at the 5' end of the intron, while the acceptor splice site is at the 3' end.</p></li>
<li><p>The function utilizes the <code>getSeq</code> function from the <code>BSgenome</code> package to extract the nucleotide sequences for both donor and acceptor sites from the specified genome (default is hg38 for humans).</p></li>
<li><p>The resulting sequences are added as new columns (<code>donor_ss</code> and <code>acceptor_ss</code>) to the original input data frame.</p></li>
<li><p>The final data frame includes the splice site sequences for each intron, allowing for analysis of splicing efficiency or identification of consensus motifs.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="extract_introns.html">extract_introns</a></code>, <code><a href="find_cryptic_splice_sites.html">find_cryptic_splice_sites</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg38</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">file_v1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"gencode.v1.example.gtf.gz"</span>, package <span class="op">=</span> <span class="st">"GencoDymo2"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gtf_v1</span> <span class="op">&lt;-</span> <span class="fu"><a href="load_file.html">load_file</a></span><span class="op">(</span><span class="va">file_v1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">introns_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="extract_introns.html">extract_introns</a></span><span class="op">(</span><span class="va">gtf_v1</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing input...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing single-exon transcripts...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Single-exon transcripts: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting intron coordinates...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Collecting intron data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total introns: 3</span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">assign_splice_sites</span><span class="op">(</span><span class="va">introns_df</span>, genome <span class="op">=</span> <span class="va">BSgenome.Hsapiens.UCSC.hg38</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Preparing donor splice sites data...</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Preparing acceptor splice sites data...</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Merging donor and acceptor splice site data...</span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Monah Abou Alezz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

