<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Extract Splice Site Motifs for MaxEntScan Analysis (5' or 3') — extract_ss_motif • GencoDymo2</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Extract Splice Site Motifs for MaxEntScan Analysis (5' or 3') — extract_ss_motif"><meta name="description" content="This function extracts splice site motifs (5' splice site (5ss) or 3' splice site (3ss)) from a genomic dataset. It retrieves the donor or acceptor splice site motifs for each intron, based on the strand orientation,
and compiles them into a FASTA file, which can be used for further analysis (e.g., MaxEntScan)."><meta property="og:description" content="This function extracts splice site motifs (5' splice site (5ss) or 3' splice site (3ss)) from a genomic dataset. It retrieves the donor or acceptor splice site motifs for each intron, based on the strand orientation,
and compiles them into a FASTA file, which can be used for further analysis (e.g., MaxEntScan)."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GencoDymo2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GencoDymo2_vignette.html">GencoDymo2</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/monahton/GencoDymo2/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Extract Splice Site Motifs for MaxEntScan Analysis (5' or 3')</h1>
      <small class="dont-index">Source: <a href="https://github.com/monahton/GencoDymo2/blob/main/R/introns.R" class="external-link"><code>R/introns.R</code></a></small>
      <div class="d-none name"><code>extract_ss_motif.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function extracts splice site motifs (5' splice site (5ss) or 3' splice site (3ss)) from a genomic dataset. It retrieves the donor or acceptor splice site motifs for each intron, based on the strand orientation,
and compiles them into a FASTA file, which can be used for further analysis (e.g., MaxEntScan).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">extract_ss_motif</span><span class="op">(</span><span class="va">input</span>, <span class="va">genome</span>, <span class="va">type</span>, <span class="va">verbose</span>, <span class="va">save_fasta</span>, <span class="va">output_file</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-input">input<a class="anchor" aria-label="anchor" href="#arg-input"></a></dt>
<dd><p>A data frame containing genomic information with the following required columns:</p><ul><li><p><code>seqnames</code>: Chromosome or scaffold names.</p></li>
<li><p><code>strand</code>: Strand orientation, either '+' or '-'.</p></li>
<li><p><code>intron_start</code>: Start position of the intron.</p></li>
<li><p><code>intron_end</code>: End position of the intron.</p></li>
<li><p><code>transcript_id</code>: Identifier for the transcript.</p></li>
<li><p><code>intron_number</code>: Identifier for the intron within the transcript.</p></li>
</ul></dd>


<dt id="arg-genome">genome<a class="anchor" aria-label="anchor" href="#arg-genome"></a></dt>
<dd><p>A genome object from the BSgenome package (default is <code>BSgenome.Hsapiens.UCSC.hg38</code>).</p></dd>


<dt id="arg-type">type<a class="anchor" aria-label="anchor" href="#arg-type"></a></dt>
<dd><p>A string indicating which splice site motif to extract. One of <code>"5ss"</code> (donor splice site) or <code>"3ss"</code> (acceptor splice site).</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Logical; if <code>TRUE</code>, progress messages will be printed. Default is <code>TRUE</code>.</p></dd>


<dt id="arg-save-fasta">save_fasta<a class="anchor" aria-label="anchor" href="#arg-save-fasta"></a></dt>
<dd><p>Logical; if <code>TRUE</code>, a FASTA file will be saved containing the extracted motifs. Default is <code>FALSE</code>.</p></dd>


<dt id="arg-output-file">output_file<a class="anchor" aria-label="anchor" href="#arg-output-file"></a></dt>
<dd><p>A string specifying the output file path and name for the FASTA file. If <code>NULL</code>, a default name will be used (either "5ss_motif_fasta.fa" or "3ss_motif_fasta.fa").</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A data frame with:</p><ul><li><p><code>donor_ss_motif</code> or <code>acceptor_ss_motif</code>: 9bp (5' ss) or 23bp (3' ss) sequence.</p></li>
<li><p>Genomic coordinates and transcript metadata.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function performs the following steps:</p><ul><li><p>Based on the <code>type</code> argument, the function prepares coordinates for extracting either donor (5ss) or acceptor (3ss) splice site motifs,
adjusting the motif start and end positions depending on the strand orientation.</p></li>
<li><p>The motif sequences are then extracted from the specified genome using the <code>getSeq</code> function from the BSgenome package.</p></li>
<li><p>If <code>save_fasta</code> is <code>TRUE</code>, a FASTA file is generated containing the extracted motifs, with transcript IDs and intron numbers
used as FASTA headers.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="assign_splice_sites.html">assign_splice_sites</a></code>, <code><a href="df_to_fasta.html">df_to_fasta</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">file_v1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"gencode.v1.example.gtf.gz"</span>, package <span class="op">=</span> <span class="st">"GencoDymo2"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gtf_v1</span> <span class="op">&lt;-</span> <span class="fu"><a href="load_file.html">load_file</a></span><span class="op">(</span><span class="va">file_v1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">introns</span> <span class="op">&lt;-</span> <span class="fu"><a href="extract_introns.html">extract_introns</a></span><span class="op">(</span><span class="va">gtf_v1</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing input...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing single-exon transcripts...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Single-exon transcripts: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting intron coordinates...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Collecting intron data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total introns: 3</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg38</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Extract donor splice site motifs</span></span></span>
<span class="r-in"><span><span class="va">motifs_df</span> <span class="op">&lt;-</span> <span class="fu">extract_ss_motif</span><span class="op">(</span><span class="va">introns</span>, <span class="va">BSgenome.Hsapiens.UCSC.hg38</span>, <span class="st">"5ss"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Extract acceptor splice site motifs without saving the FASTA file</span></span></span>
<span class="r-in"><span><span class="va">motifs_df</span> <span class="op">&lt;-</span> <span class="fu">extract_ss_motif</span><span class="op">(</span><span class="va">introns</span>, <span class="va">BSgenome.Hsapiens.UCSC.hg38</span>, type <span class="op">=</span> <span class="st">"3ss"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Monah Abou Alezz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

