<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Identify Potential Cryptic Splice Sites. — find_cryptic_splice_sites • GencoDymo2</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Identify Potential Cryptic Splice Sites. — find_cryptic_splice_sites"><meta name="description" content="This function identifies potential cryptic splice sites by comparing sequence motifs in introns to canonical splice site motifs (donor and acceptor). Cryptic splice sites are those that do not match the canonical donor (GT) or acceptor motifs (AG). It compares the identified splice sites with the provided canonical motifs and flags the sites that differ from the canonical patterns, making it useful for studying aberrant splicing events."><meta property="og:description" content="This function identifies potential cryptic splice sites by comparing sequence motifs in introns to canonical splice site motifs (donor and acceptor). Cryptic splice sites are those that do not match the canonical donor (GT) or acceptor motifs (AG). It compares the identified splice sites with the provided canonical motifs and flags the sites that differ from the canonical patterns, making it useful for studying aberrant splicing events."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GencoDymo2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GencoDymo2_vignette.html">GencoDymo2: Comprehensive Analysis of GENCODE Annotations and Splice Site Motifs</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/monahton/GencoDymo2/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Identify Potential Cryptic Splice Sites.</h1>
      <small class="dont-index">Source: <a href="https://github.com/monahton/GencoDymo2/blob/v1.0.2/R/introns.R" class="external-link"><code>R/introns.R</code></a></small>
      <div class="d-none name"><code>find_cryptic_splice_sites.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function identifies potential cryptic splice sites by comparing sequence motifs in introns to canonical splice site motifs (donor and acceptor). Cryptic splice sites are those that do not match the canonical donor (GT) or acceptor motifs (AG). It compares the identified splice sites with the provided canonical motifs and flags the sites that differ from the canonical patterns, making it useful for studying aberrant splicing events.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">find_cryptic_splice_sites</span><span class="op">(</span><span class="va">input</span>, <span class="va">genome</span>, <span class="va">canonical_donor</span>, <span class="va">canonical_acceptor</span>, <span class="va">verbose</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-input">input<a class="anchor" aria-label="anchor" href="#arg-input"></a></dt>
<dd><p>A data frame containing intron coordinates, ideally generated
by <code><a href="extract_introns.html">extract_introns()</a></code> and <code><a href="assign_splice_sites.html">assign_splice_sites()</a></code>. Must contain columns: <code>seqnames</code>, <code>intron_start</code>,
<code>intron_end</code>, <code>strand</code>, <code>transcript_id</code>, <code>intron_number</code>, <code>gene_name</code>,
<code>gene_id</code>, <code>donor_ss</code> and <code>acceptor_ss</code>.</p></dd>


<dt id="arg-genome">genome<a class="anchor" aria-label="anchor" href="#arg-genome"></a></dt>
<dd><p>A BSgenome object representing the genome sequence.
This is used to extract the sequence for each intron to identify splice sites.</p></dd>


<dt id="arg-canonical-donor">canonical_donor<a class="anchor" aria-label="anchor" href="#arg-canonical-donor"></a></dt>
<dd><p>A character vector of canonical donor splice site motifs.
Default is <code>c("GT")</code>.</p></dd>


<dt id="arg-canonical-acceptor">canonical_acceptor<a class="anchor" aria-label="anchor" href="#arg-canonical-acceptor"></a></dt>
<dd><p>A character vector of canonical acceptor splice site motifs.
Default is <code>c("AG")</code>.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Logical; if <code>TRUE</code>, progress messages are printed. Default is <code>TRUE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>The input data frame with two logical columns:</p><ul><li><p><code>cryptic_donor</code>: <code>TRUE</code> if donor site is non-canonical.</p></li>
<li><p><code>cryptic_acceptor</code>: <code>TRUE</code> if acceptor site is non-canonical.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function performs the following steps:</p><ul><li><p>It assigns donor and acceptor splice sites to each intron using the <code>assign_splice_sites</code> function.</p></li>
<li><p>It compares the identified donor and acceptor splice sites against the provided canonical motifs (<code>GT</code> for donor and <code>AG</code> for acceptor by default). If the splice site sequences do not match the canonical motifs, they are flagged as cryptic.</p></li>
<li><p>The function returns a data frame with the same intron information, including additional columns <code>cryptic_donor</code> and <code>cryptic_acceptor</code> indicating whether the splice sites are cryptic.</p></li>
<li><p>The progress of the function is printed if the <code>verbose</code> argument is set to <code>TRUE</code>, showing also the total number of cryptic donor and acceptor sites and their respective percentages.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="assign_splice_sites.html">assign_splice_sites</a></code>, <code><a href="extract_ss_motif.html">extract_ss_motif</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BSgenome.Hsapiens.UCSC.hg38"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">file_v1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"gencode.v1.example.gtf.gz"</span>, package <span class="op">=</span> <span class="st">"GencoDymo2"</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">gtf_v1</span> <span class="op">&lt;-</span> <span class="fu"><a href="load_file.html">load_file</a></span><span class="op">(</span><span class="va">file_v1</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">introns_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="extract_introns.html">extract_introns</a></span><span class="op">(</span><span class="va">gtf_v1</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">introns_ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="assign_splice_sites.html">assign_splice_sites</a></span><span class="op">(</span><span class="va">introns_df</span>, genome <span class="op">=</span> <span class="va">BSgenome.Hsapiens.UCSC.hg38</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">cryptic_sites</span> <span class="op">&lt;-</span> <span class="fu">find_cryptic_splice_sites</span><span class="op">(</span><span class="va">introns_ss</span>, <span class="va">BSgenome.Hsapiens.UCSC.hg38</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cryptic_sites</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">}</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Monah Abou Alezz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

