<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>GencoDymo2: Comprehensive Analysis of GENCODE Annotations and Splice Site Motifs • GencoDymo2</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="GencoDymo2: Comprehensive Analysis of GENCODE Annotations and Splice Site Motifs">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GencoDymo2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/GencoDymo2_vignette.html">GencoDymo2: Comprehensive Analysis of GENCODE Annotations and Splice Site Motifs</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/monahton/GencoDymo2/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>GencoDymo2: Comprehensive Analysis of GENCODE Annotations and Splice Site Motifs</h1>
                        <h4 data-toc-skip class="author">Monah Abou
Alezz<sup>1</sup>, Lorenzo Salviati<sup>2</sup>, Roberta
Alfieri<sup>3</sup>, Silvia Bione<sup>4</sup>
</h4>
            <address class="author_afil">
      <sup>1</sup>San Raffaele Telethon Institute for Gene Therapy,
Istituto di Ricovero e Cura a Carattere Scientifico (IRCCS) San Raffaele
Scientific Institute, 20132 Milan, Italy<br><sup>2</sup>Human
Technopole, Viale Rita Levi-Montalcini 1, 20157 Milan, Italy<br><sup>3</sup>National Research Council, Institute for Biomedical
Technologies, 20054 Segrate, Italy<br><sup>4</sup>Computational Biology
Unit, Institute of Molecular Genetics Luigi Luca Cavalli-Sforza,
National Research Council, 27100 Pavia, Italy<br><h4 data-toc-skip class="date">2025-07-14</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/monahton/GencoDymo2/blob/main/vignettes/GencoDymo2_vignette.Rmd" class="external-link"><code>vignettes/GencoDymo2_vignette.Rmd</code></a></small>
      <div class="d-none name"><code>GencoDymo2_vignette.Rmd</code></div>
    </address>
</div>

    
    
<style>
a:link {
    color: red;
}

a:visited {
    color: green;
}

a:hover {
    color: hotpink;
}
p.caption {
  font-size: 0.8em;
}

</style>
<hr>
<p><br></p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr>
<p>Gene annotations are provided by several databases that integrate
data from systematic explorations of various experimental and
computational resources. The <strong>GENCODE</strong> project [1], part
of the ENCODE project [2], offers accurate annotations of the human and
mouse genomes derived from literature, primary data repositories,
computational predictions, manual annotation, and experimental
validation of genes and transcripts, including noncoding transcripts
that are continually discovered and annotated. GENCODE is one of the
most comprehensive and standardized databases for gene annotations,
widely used by the scientific community.​</p>
<p>The gene sets provided by GENCODE are comprehensive and include
protein-coding and non-coding loci, encompassing alternatively spliced
isoforms and pseudogenes. GENCODE gene annotations are regularly updated
and released as the Ensembl/GENCODE gene sets, accessible via the
official website <a href="https://www.gencodegenes.org" class="external-link uri">https://www.gencodegenes.org</a>. As of October 2024, the
latest human release is GENCODE 47, and the latest mouse release is
GENCODE M36 [1]. GENCODE gene sets are released approximately four times
a year for mouse and twice a year for human [1].​</p>
<p><strong>GencoDymo2</strong> is an R package designed to interrogate
different releases of GENCODE annotations from the human and mouse
genomes. It provides a streamlined interface for accessing and
processing GENCODE annotations, providing easily accessible data
regarding annotation statistics, release comparisons, and information on
introns and splice sites. Moreover, GencoDymo2 can produce FASTA files
of donor and acceptor splice site motifs that can be directly uploaded
to the MaxEntScan tool [3] for the calculation of splice site
scores.</p>
<p><br></p>
<p>GencoDymo2 can be used to import and process any gtf/gff3 formatted
file. It automatizes and speeds up the following data manipulation
steps:</p>
<ul>
<li>Dynamic detection of latest GENCODE releases from the official
website</li>
<li>Unified interface for GTF/GFF3 file retrieval for the human and
mouse genomes</li>
<li>Compare annotations from different GENCODE releases</li>
<li>Perform summary statistics on gene annotations</li>
<li>Extract introns information</li>
<li>Assign splice sites consensus sequence</li>
<li>Compile MaxEntScan splice sites motifs</li>
</ul>
<hr>
<p><br></p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<hr>
<p>GencoDymo2 runs in the R statistical computing environment. You will
need R version 4.1.0 or higher. The most recent version of GencoDymo2
can be installed from GitHub. We recommend using the pak package for
fast and reliable installation.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install pak if not already installed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://pak.r-lib.org/" class="external-link">"pak"</a></span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"pak"</span><span class="op">)</span></span>
<span><span class="co"># Install from GitHub</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"github::monahton/GencoDymo2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/monahton/GencoDymo2" class="external-link">GencoDymo2</a></span><span class="op">)</span></span></code></pre></div>
<hr>
<p><br></p>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="latest-releases">Latest Releases<a class="anchor" aria-label="anchor" href="#latest-releases"></a>
</h3>
<p>GENCODE releases annotations approximately every three months,
coinciding with Ensembl releases. The latest release from GENCODE for
human and mouse genomes can be obtained using the <span style="color:coral"><code><a href="../reference/get_latest_release.html">get_latest_release()</a></code></span> function.
This function queries the official GENCODE website to determine the most
recent release for a given species (human or mouse).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Fetch the most recent human and mouse GENCODE release identifiers</span></span>
<span><span class="va">human_release</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_latest_release.html">get_latest_release</a></span><span class="op">(</span><span class="st">"human"</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">mouse_release</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_latest_release.html">get_latest_release</a></span><span class="op">(</span><span class="st">"mouse"</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Latest human GENCODE release: release_47</span></span></code></pre>
<pre><code><span><span class="co">## Latest human GENCODE release: release_M36</span></span></code></pre>
<p><br></p>
<blockquote>
<p><strong><em>NOTE:</em></strong> Behind the scenes,
get_latest_release() parses the Ensembl public FTP site, extracts the
release version (e.g., “release_47”), and returns it for programmatic
use.</p>
</blockquote>
<p><br></p>
</div>
<div class="section level3">
<h3 id="download-gtfgff3-files">Download GTF/GFF3 Files<a class="anchor" aria-label="anchor" href="#download-gtfgff3-files"></a>
</h3>
<p>GencoDymo2 allows the download of <code>gtf</code> and
<code>gff3</code> files of the latest release or any release specified
by the user. These files can be obtained for human and mouse genome and
can be saved in a path specified by the user. With the release
identifiers in hand, we can download these annotation files:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Download latest human long noncoding RNAs GTF</span></span>
<span><span class="va">lnc_47_gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_gtf.html">get_gtf</a></span><span class="op">(</span></span>
<span>  species <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  release_version <span class="op">=</span> <span class="va">human_release</span>,</span>
<span>  annotation_type <span class="op">=</span> <span class="st">"long_noncoding_RNAs.gtf.gz"</span>,</span>
<span>  dest_folder <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Download previous human release (release_46) for comparison</span></span>
<span><span class="va">lnc_46_gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_gtf.html">get_gtf</a></span><span class="op">(</span></span>
<span>  species <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  release_version <span class="op">=</span> <span class="st">"release_46"</span>,</span>
<span>  annotation_type <span class="op">=</span> <span class="st">"long_noncoding_RNAs.gtf.gz"</span>,</span>
<span>  dest_folder <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Download latest mouse primary assembly annotations (GFF3)</span></span>
<span><span class="va">mouse_36_gff3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_gff3.html">get_gff3</a></span><span class="op">(</span></span>
<span>  species <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>  release_version <span class="op">=</span> <span class="va">mouse_release</span>,</span>
<span>  annotation_type <span class="op">=</span> <span class="st">"primary_assembly.annotation.gff3.gz"</span>,</span>
<span>  dest_folder <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="supported-annotation-types">Supported Annotation Types<a class="anchor" aria-label="anchor" href="#supported-annotation-types"></a>
</h4>
<p>The valid annotation types for both the gtf and gff3 format can be
one of the following:</p>
<pre><code><span><span class="co">## Valid Annotation Types:</span></span></code></pre>
<pre><code><span><span class="co">## [1] "annotation"                           </span></span>
<span><span class="co">## [2] "basic.annotation"                     </span></span>
<span><span class="co">## [3] "chr_patch_hapl_scaff.annotation"      </span></span>
<span><span class="co">## [4] "chr_patch_hapl_scaff.basic.annotation"</span></span>
<span><span class="co">## [5] "long_noncoding_RNAs"                  </span></span>
<span><span class="co">## [6] "primary_assembly.annotation"          </span></span>
<span><span class="co">## [7] "primary_assembly.basic.annotation"    </span></span>
<span><span class="co">## [8] "tRNAs"                                </span></span>
<span><span class="co">## [9] "polyAs"</span></span></code></pre>
<p><br></p>
</div>
</div>
<div class="section level3">
<h3 id="load-and-inspect">Load and Inspect<a class="anchor" aria-label="anchor" href="#load-and-inspect"></a>
</h3>
<p>A <code>gtf</code> or <code>gff3</code> file can be imported to R as
a dataframe using the <span style="color:coral"><code><a href="../reference/load_file.html">load_file()</a></code></span> function. Once
downloaded, load the files into R as data frames using load_file(). This
function automatically detects GTF vs GFF3 and parses the attributes
into columns.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Loading using the stored paths from previous steps</span></span>
<span><span class="va">lnc_47_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_file.html">load_file</a></span><span class="op">(</span><span class="va">lnc_47_gtf</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lnc_47_df</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Alternatively, specify the file path directly</span></span>
<span><span class="va">lnc_46_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_file.html">load_file</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"gencode.v46.long_noncoding_RNAs.gtf.gz"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lnc_46_df</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load mouse GFF3</span></span>
<span><span class="va">mouse_pri_36</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_file.html">load_file</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">"gencode.vM36.primary_assembly.annotation.gff3.gz"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mouse_pri_36</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level3">
<h3 id="compare-releases">Compare Releases<a class="anchor" aria-label="anchor" href="#compare-releases"></a>
</h3>
<p>As GENCODE regularly updates its annotations, different releases from
GENCODE contain differing number of annotations. Some genes are added in
new releases while others are classified into different gene classes. In
the majority of cases, the Ensembl gene ID of genes remains constant
while the gene name might vary.<br>
To quantify changes between two releases, use <span style="color:coral"><code><a href="../reference/compare_release.html">compare_release()</a></code></span>. Specify the
two annotation data frames and the type of feature to compare (“gene”,
“transcript”, “exon”, etc.). Additional filters like the gene_type and
the baseline can allow to focus on subsets (e.g., only TEC or
protein_coding genes) and different reference counts for the calculation
of the difference percentage.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compare gene counts between release 47 and 46</span></span>
<span><span class="va">gene_comparison</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compare_release.html">compare_release</a></span><span class="op">(</span><span class="va">lnc_47_df</span>, <span class="va">lnc_46_df</span>, type <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare exon counts</span></span>
<span><span class="va">exon_comparison</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compare_release.html">compare_release</a></span><span class="op">(</span><span class="va">lnc_47_df</span>, <span class="va">lnc_46_df</span>, type <span class="op">=</span> <span class="st">"exon"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare a specific gene biotype (e.g., TEC) using a custom baseline</span></span>
<span><span class="va">comparison</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compare_release.html">compare_release</a></span><span class="op">(</span></span>
<span>  <span class="va">lnc_47_df</span>,</span>
<span>  <span class="va">lnc_46_df</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"gene"</span>,</span>
<span>  gene_type <span class="op">=</span> <span class="st">"TEC"</span>,</span>
<span>  baseline <span class="op">=</span> <span class="st">"count1"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The function produces as an output a <em><strong>Delta</strong></em>
value, that corresponds to the difference in number between the
specified elements of the two inputs, in addition to its
<strong><em>Percentage</em></strong>.</p>
<p><br></p>
<blockquote>
<p><strong><em>NOTE:</em></strong> The output is a data frame
summarizing absolute and percentage differences in feature counts,
enabling rapid assessment of annotation growth or refinement.</p>
</blockquote>
<p><br></p>
</div>
<div class="section level3">
<h3 id="extract-introns-coordinates">Extract Introns Coordinates<a class="anchor" aria-label="anchor" href="#extract-introns-coordinates"></a>
</h3>
<p>The gtf files provided by GENCODE list genes, transcripts, and exons
while intronic regions must be inferred. Given the loaded data frame
from the gtf files of GENCODE, the <span style="color:coral"><code><a href="../reference/extract_introns.html">extract_introns()</a></code></span> function
computes intron coordinates for each transcript by sorting exon ranges
and identifying gaps. It returns a dataframe containing introns
coordinates and position within each transcript.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Human lncRNA introns for release 47</span></span>
<span><span class="va">introns_lnc_47</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_introns.html">extract_introns</a></span><span class="op">(</span><span class="va">lnc_47_df</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Mouse introns (filtering to primary chromosomes first)</span></span>
<span><span class="va">mouse_pri_36</span> <span class="op">&lt;-</span> <span class="va">mouse_pri_36</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"^chr"</span>, <span class="va">mouse_pri_36</span><span class="op">$</span><span class="va">seqnames</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">mouse_introns_pri_36</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_introns.html">extract_introns</a></span><span class="op">(</span><span class="va">mouse_pri_36</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level3">
<h3 id="extract-donor-and-acceptor-splice-sites">Extract Donor and Acceptor Splice Sites<a class="anchor" aria-label="anchor" href="#extract-donor-and-acceptor-splice-sites"></a>
</h3>
<p>Splice sites are defined by dinucleotide motifs at intron boundaries.
At the 5’ end, the donor site includes an almost invariant sequence GT
(GU in the RNA). At the 3’ end, the intron terminates with the acceptor
site, an almost invariant AG sequence.</p>
<p><br></p>
<div class="float">
<img src="Consensus_intron.jpg" alt="Fig1. Donor and acceptor splice sites consensus. Image taken from (https://commons.wikimedia.org/wiki/File:Consensus_intron.jpg)."><div class="figcaption">Fig1. Donor and acceptor splice sites consensus.
Image taken from (<a href="https://commons.wikimedia.org/wiki/File:Consensus_intron.jpg" class="external-link uri">https://commons.wikimedia.org/wiki/File:Consensus_intron.jpg</a>).</div>
</div>
<p><br></p>
<p>The majority of introns belong to the U2-type spliceosome and are
flanked by GT– AG splice site dinucleotides. The most frequent exception
to this rule are the U2-type GC–AG splice sites, comprising ∼0.8% of
human splice sites and about ∼0.09% of the human splice sites belong to
the U12-type which are processed by the minor spliceosome and are
flanked by AT–AC dinucleotides [4].<br>
The <span style="color:coral"><code><a href="../reference/assign_splice_sites.html">assign_splice_sites()</a></code></span>
function retrieves these flanking sequences from a BSgenome object. It
adds two new columns to the data frame containing the introns
coordinates, assigning both the 5’ and 3’ splice sites. Sequences should
be retrieved from a loaded BSgenome object such as
<code>BSgenome.Hsapiens.UCSC.hg38</code> for human and
<code>BSgenome.Mmusculus.UCSC.mm39</code> for mouse.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Human</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg38</span><span class="op">)</span></span>
<span><span class="va">lnc_47_ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assign_splice_sites.html">assign_splice_sites</a></span><span class="op">(</span></span>
<span>  <span class="va">introns_lnc_47</span>,</span>
<span>  genome <span class="op">=</span> <span class="va">BSgenome.Hsapiens.UCSC.hg38</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">T</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Mouse</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Mmusculus.UCSC.mm39</span><span class="op">)</span></span>
<span><span class="va">mouse_pri_36_ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assign_splice_sites.html">assign_splice_sites</a></span><span class="op">(</span></span>
<span>  <span class="va">mouse_introns_pri_36</span>,</span>
<span>  genome <span class="op">=</span> <span class="va">BSgenome.Mmusculus.UCSC.mm39</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">T</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><br></p>
<div class="section level4">
<h4 id="identify-cryptic-splice-sites">Identify Cryptic splice sites<a class="anchor" aria-label="anchor" href="#identify-cryptic-splice-sites"></a>
</h4>
<p>Cryptic splice sites are those that do not match the canonical donor
(GT) or acceptor motifs (AG). The <span style="color:coral"><code><a href="../reference/find_cryptic_splice_sites.html">find_cryptic_splice_sites()</a></code></span>
function identifies potential cryptic splice sites by comparing sequence
motifs in introns to canonical splice site motifs (donor and acceptor)
that can be also specified by the user. It compares the identified
splice sites with the provided canonical motifs and flags the sites that
differ from the canonical patterns, making it useful for studying
aberrant splicing events.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Identify cryptic (non-canonical) splice sites</span></span>
<span><span class="va">cryptic_ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find_cryptic_splice_sites.html">find_cryptic_splice_sites</a></span><span class="op">(</span></span>
<span>  <span class="va">lnc_47_ss</span>,</span>
<span>  genome <span class="op">=</span> <span class="va">BSgenome.Hsapiens.UCSC.hg38</span>,</span>
<span>  canonical_donor <span class="op">=</span> <span class="st">"GT"</span>,</span>
<span>  canonical_acceptor <span class="op">=</span> <span class="st">"AG"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level4">
<h4 id="extract-splice-sites-motifs-for-maxentscan-webtool">Extract splice sites motifs for MaxEntScan webtool<a class="anchor" aria-label="anchor" href="#extract-splice-sites-motifs-for-maxentscan-webtool"></a>
</h4>
<p><strong>MaxEntScan</strong> is a webtool based on the approach for
modeling the sequences of short sequence motifs involved in RNA splicing
which simultaneously accounts for non-adjacent as well as adjacent
dependencies between positions. This method is based on the <em>‘Maximum
Entropy Principle’</em> and generalizes most previous probabilistic
models of sequence motifs such as weight matrix models and inhomogeneous
Markov models [3].<br>
The <strong>MaxEntScan::score5ss</strong> assign scores for donor splice
sites according to four models by using <strong>9-mer SEQUENCES</strong>
motif as input in a <code>.fa file</code>. The 9-mers motif contain 3
bases in the exon and 6 bases in the intron.<br>
The <strong>MaxEntScan::score3ss</strong> assign scores for donor splice
sites according to three models. It takes as input a
<code>.fa file</code> of <strong>23-mer SEQUENCES</strong> in which 20
bases are in the intron and 3 bases in the exon.<br>
The <span style="color:coral"><code><a href="../reference/extract_ss_motif.html">extract_ss_motif()</a></code></span>
function is used along with BSgenome object of the studied species to
retrieve <em>MaxEntScan::score5ss</em> and <em>MaxEntScan::score3ss</em>
motif sequences.<br>
The user can choose to generate a <strong>fasta file</strong> in the
working directory or any specified path as an output, that contains
either 9-mers or 23-mers, respectively for each 5’ and 3’ splice-sites.
It generates also a <strong>dataframe</strong> for the coordinates and
IDs of the corresponding motifs.<br>
The generated fasta files can then be directly utilized by
<em>MaxEntScan</em> tools.<br>
The first argument <code>input</code> is a dataframe containing the
intron coordinates. The second argument <code>genome</code> is a
BSgenome object (the human genome sequence hg38 is used by default). The
third argument <code>type</code> specifies whether to extract motifs at
the 5ss or the 3ss. Users can optionally set the argument
<code>save_fasta</code> to TRUE to and the <code>output_file</code>
argument to save a fasta file.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Donor motifs (5'ss)</span></span>
<span><span class="va">motifs_donor</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_ss_motif.html">extract_ss_motif</a></span><span class="op">(</span></span>
<span>  input <span class="op">=</span> <span class="va">lnc_47_ss</span>,</span>
<span>  genome <span class="op">=</span> <span class="va">BSgenome.Hsapiens.UCSC.hg38</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"5ss"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  save_fasta <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  output_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"lnc_47_5ss_motifs.fa"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Acceptor motifs (3'ss)</span></span>
<span><span class="va">motifs_acc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_ss_motif.html">extract_ss_motif</a></span><span class="op">(</span></span>
<span>  input <span class="op">=</span> <span class="va">lnc_47_ss</span>,</span>
<span>  genome <span class="op">=</span> <span class="va">BSgenome.Hsapiens.UCSC.hg38</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"3ss"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  save_fasta <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  output_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"lnc_47_3ss_motifs.fa"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<hr>
<p><br></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="additional-utilities">Additional Utilities<a class="anchor" aria-label="anchor" href="#additional-utilities"></a>
</h2>
<hr>
<div class="section level4">
<h4 id="single-exon-features">Single-Exon Features<a class="anchor" aria-label="anchor" href="#single-exon-features"></a>
</h4>
<p>Some genes are unspliced: composed by one single exon and have no
introns.<br>
The <span style="color:coral"><code><a href="../reference/extract_single_exon.html">extract_single_exon()</a></code></span>
function can extract all the <strong><em>single-exon</em></strong> genes
or transcripts and returns in a dataframe with their ensembl gene IDs or
transcript IDs.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## identify single exon genes and transcripts</span></span>
<span><span class="va">single_exon_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_single_exon.html">extract_single_exon</a></span><span class="op">(</span><span class="va">lnc_47_df</span>, level <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span></span>
<span><span class="va">single_exon_trans</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_single_exon.html">extract_single_exon</a></span><span class="op">(</span><span class="va">lnc_47_df</span>, level <span class="op">=</span> <span class="st">"transcript"</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level4">
<h4 id="exon-classification">Exon Classification<a class="anchor" aria-label="anchor" href="#exon-classification"></a>
</h4>
<p>The <span style="color:coral"><code><a href="../reference/classify_exons.html">classify_exons()</a></code></span>
adds a new column <code>(EXON_CLASSIFICATION)</code> to the input
dataframe, describing the position of each exon. It labels each exon as
<em>(first_exons)</em>, <em>(inner_exons)</em>, <em>(last_exons)</em>
and <em>(single_exons)</em> facilitating position-based analyses. This
classification is informative considering that first and last exons
often have peculiar regulative roles.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Assign the ordinal position of exons</span></span>
<span><span class="va">lnc_47_class_exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/classify_exons.html">classify_exons</a></span><span class="op">(</span><span class="va">lnc_47_df</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level4">
<h4 id="spliced-transcript-length">Spliced Transcript Length<a class="anchor" aria-label="anchor" href="#spliced-transcript-length"></a>
</h4>
<p>The <span style="color:coral"><code><a href="../reference/spliced_trans_length.html">spliced_trans_length()</a></code></span> function
computes the mature (post-splicing) transcript length by summing exon
widths per transcript.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Length of spliced transcript  </span></span>
<span><span class="va">lnc_47_spliced_length</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spliced_trans_length.html">spliced_trans_length</a></span><span class="op">(</span><span class="va">lnc_47_df</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lnc_47_spliced_length</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level4">
<h4 id="summary-statistics">Summary Statistics<a class="anchor" aria-label="anchor" href="#summary-statistics"></a>
</h4>
<p>GencoDymo2 provides the <span style="color:coral"><code><a href="../reference/stat_summary.html">stat_summary()</a></code></span> function that
gives a brief summary of data annotations in a particular GENCODE
release regarding genes, transcripts, exons and introns.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Exon length statistics</span></span>
<span><span class="va">lnc_47_exon_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/stat_summary.html">stat_summary</a></span><span class="op">(</span><span class="va">lnc_47_class_exons</span>, type <span class="op">=</span> <span class="st">"exon"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Intron length statistics</span></span>
<span><span class="va">lnc_47_intron_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/stat_summary.html">stat_summary</a></span><span class="op">(</span><span class="va">introns_lnc_47</span>, type <span class="op">=</span> <span class="st">"intron"</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level4">
<h4 id="gc-content-calculation">GC Content Calculation<a class="anchor" aria-label="anchor" href="#gc-content-calculation"></a>
</h4>
<p>The <span style="color:coral"><code><a href="../reference/calculate_gc_content.html">calculate_gc_content()</a></code></span> function
computes GC percentage for any feature set using a BSgenome
reference.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Human</span></span>
<span><span class="va">lnc_47_gc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_gc_content.html">calculate_gc_content</a></span><span class="op">(</span></span>
<span>  <span class="va">lnc_47_df</span>,</span>
<span>  genome <span class="op">=</span> <span class="va">BSgenome.Hsapiens.UCSC.hg38</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co"># Mouse</span></span>
<span><span class="va">mouse_pri_36_gc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_gc_content.html">calculate_gc_content</a></span><span class="op">(</span></span>
<span>  <span class="va">mouse_pri_36</span>,</span>
<span>  genome <span class="op">=</span> <span class="va">BSgenome.Mmusculus.UCSC.mm39</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level4">
<h4 id="extract-cds-sequences">Extract CDS Sequences<a class="anchor" aria-label="anchor" href="#extract-cds-sequences"></a>
</h4>
<p>For coding annotations, <span style="color:coral"><code><a href="../reference/extract_cds_sequences.html">extract_cds_sequences()</a></code></span> writes
CDS FASTA files from GRanges objects.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="co"># Convert to GRanges and extract</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>mouse_pri_36_granges <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(mouse_pri_36)</span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a>mouse_cds_seqs <span class="ot">&lt;-</span> <span class="fu">extract_cds_sequences</span>(</span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a>  mouse_pri_36_granges,</span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a>  BSgenome.Mmusculus.UCSC.mm39,</span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a>  <span class="at">save_fasta =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a>  <span class="at">output_file =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(), <span class="st">"mouse_pri_36_CDS.fa.gz"</span>)</span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a>)</span></code></pre></div>
<hr>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<hr>
<p>The following provides the session information used when compiling
this document.</p>
<pre><code><span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">##  version  R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">##  os       Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">##  system   x86_64, linux-gnu</span></span>
<span><span class="co">##  ui       X11</span></span>
<span><span class="co">##  language en-US</span></span>
<span><span class="co">##  collate  C.UTF-8</span></span>
<span><span class="co">##  ctype    C.UTF-8</span></span>
<span><span class="co">##  tz       UTC</span></span>
<span><span class="co">##  date     2025-07-14</span></span>
<span><span class="co">##  pandoc   3.1.11 @ /opt/hostedtoolcache/pandoc/3.1.11/x64/ (via rmarkdown)</span></span>
<span><span class="co">##  quarto   NA</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">package    </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version</span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">##  bslib         0.9.0   <span style="color: #555555;">2025-01-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  cachem        1.1.0   <span style="color: #555555;">2024-05-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  cli           3.6.5   <span style="color: #555555;">2025-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  desc          1.4.3   <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  devtools      2.4.5   <span style="color: #555555;">2022-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  digest        0.6.37  <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  ellipsis      0.3.2   <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  evaluate      1.0.4   <span style="color: #555555;">2025-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  fastmap       1.2.0   <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  fs            1.6.6   <span style="color: #555555;">2025-04-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  glue          1.8.0   <span style="color: #555555;">2024-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  htmltools     0.5.8.1 <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  htmlwidgets   1.6.4   <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  httpuv        1.6.16  <span style="color: #555555;">2025-04-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  jquerylib     0.1.4   <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  jsonlite      2.0.0   <span style="color: #555555;">2025-03-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  knitr         1.50    <span style="color: #555555;">2025-03-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  later         1.4.2   <span style="color: #555555;">2025-04-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  lifecycle     1.0.4   <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  magrittr      2.0.3   <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  memoise       2.0.1   <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  mime          0.13    <span style="color: #555555;">2025-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  miniUI        0.1.2   <span style="color: #555555;">2025-04-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  pkgbuild      1.4.8   <span style="color: #555555;">2025-05-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  pkgdown       2.1.3   <span style="color: #555555;">2025-05-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  pkgload       1.4.0   <span style="color: #555555;">2024-06-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  profvis       0.4.0   <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  promises      1.3.3   <span style="color: #555555;">2025-05-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  purrr         1.1.0   <span style="color: #555555;">2025-07-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  R6            2.6.1   <span style="color: #555555;">2025-02-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  ragg          1.4.0   <span style="color: #555555;">2025-04-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  Rcpp          1.1.0   <span style="color: #555555;">2025-07-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  remotes       2.5.0   <span style="color: #555555;">2024-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  rlang         1.1.6   <span style="color: #555555;">2025-04-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  rmarkdown     2.29    <span style="color: #555555;">2024-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  sass          0.4.10  <span style="color: #555555;">2025-04-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  sessioninfo   1.2.3   <span style="color: #555555;">2025-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  shiny         1.11.1  <span style="color: #555555;">2025-07-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  systemfonts   1.2.3   <span style="color: #555555;">2025-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  textshaping   1.0.1   <span style="color: #555555;">2025-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  urlchecker    1.0.1   <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  usethis       3.1.0   <span style="color: #555555;">2024-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  vctrs         0.6.5   <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  xfun          0.52    <span style="color: #555555;">2025-04-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  xtable        1.8-4   <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  yaml          2.3.10  <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #555555;"> [1] /home/runner/work/_temp/Library</span></span></span>
<span><span class="co">## <span style="color: #555555;"> [2] /opt/R/4.5.1/lib/R/site-library</span></span></span>
<span><span class="co">## <span style="color: #555555;"> [3] /opt/R/4.5.1/lib/R/library</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
<hr>
<p><br></p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<hr>
<ol style="list-style-type: decimal">
<li>Frankish, A., Diekhans, M., Ferreira, A.M., et al. (2023).
<em>GENCODE: reference annotation for the human and mouse genomes in
2023</em>. Nucleic Acids Research, 51(D1), D942–D949.<br>
</li>
<li>ENCODE Project Consortium. (2012). <em>An integrated encyclopedia of
DNA elements in the human genome</em>. Nature, 489(7414), 57–74.<br>
</li>
<li>Yeo, G., &amp; Burge, C.B. (2004). <em>Maximum entropy modeling of
short sequence motifs with applications to RNA splicing signals</em>.
Journal of Computational Biology, 11(2-3), 377–394.<br>
</li>
<li>Parada GE, Munita R, Cerda CA and Gysling K. <em>“A comprehensive
survey of non-canonical splice sites in the human transcriptome”</em>.
Nucleic Acids Res. 2014. 42: 10564-10578.<br>
</li>
<li>Abou Alezz, M., Celli, L., Belotti, G., et al. (2020). <em>GC-AG
Introns Features in Long Non-coding and Protein-Coding Genes Suggest
Their Role in Gene Expression Regulation</em>. Front. Genet. 2020
11:488.</li>
</ol>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Monah Abou Alezz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
